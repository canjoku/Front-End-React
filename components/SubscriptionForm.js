import { Component } from 'react'


class SubscriptionForm extends Component {
  constructor(props){
    super(props)
    this.submitSubscriber = this.submitSubscriber.bind(this)
    this.state = {
      subscribed: false
    }
  }

  submitSubscriber (e, mutation) {
    e.preventDefault()
    const form = e.target
    mutation({
      variables: {
        subscriber: {
          email: form.email.value
        }
      }
    }).
    then(({data}) => {this.setState({subscribed: data.createSubscriber.email})}).
    catch(({error})=>{console.log('got error',error)})
    form.reset()
  }

  showSubscribeStatus () {
    if (this.state.subscribed) {
      return (
        <div>Your email {this.state.subscribed} has been succcesfully submitted..</div>
      )
    }
  }
  render () {
    const { mutation } = this.props
    return (
      <div>
        <form onSubmit={(e) => this.submitSubscriber(e, mutation)}>
          <input name="email" placeholder="Enter Valid Email"/>
          <button>Subscribe</button>
        </form>
        <div>
          {this.showSubscribeStatus()}
        </div>
      </div>
    )
  }
}

export default SubscriptionForm