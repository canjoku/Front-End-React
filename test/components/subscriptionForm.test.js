import SubscriptionForm from '../../components/SubscriptionForm'

test('<SubscriptionForm /> renders correctly', () => {
  const component = shallow(<SubscriptionForm />)
  expect(component).toMatchSnapshot()
})
