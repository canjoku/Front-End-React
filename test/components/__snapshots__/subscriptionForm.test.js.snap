// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubscriptionForm /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubscriptionForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
        >
          <input
            name="email"
            placeholder="Enter Valid Email"
          />
          <button>
            Subscribe
          </button>
        </form>,
        <div />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="email"
              placeholder="Enter Valid Email"
            />,
            <button>
              Subscribe
            </button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "placeholder": "Enter Valid Email",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Subscribe",
            },
            "ref": null,
            "rendered": "Subscribe",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
          >
            <input
              name="email"
              placeholder="Enter Valid Email"
            />
            <button>
              Subscribe
            </button>
          </form>,
          <div />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="email"
                placeholder="Enter Valid Email"
              />,
              <button>
                Subscribe
              </button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "placeholder": "Enter Valid Email",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Subscribe",
              },
              "ref": null,
              "rendered": "Subscribe",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
